node/node.go

```diff
@@ -801,3 +804,31 @@ func (n *Node) apis() []rpc.API {
 		},
 	}
 }
+
+func NtpCheckWithLocal(n *Node) error {
+	// Skip check if server is empty (e.g. `ntp.disable` flag)
+	if n.config.NtpRemoteServer == "" {
+		return nil
+	}
+
+	url, port, err := net.SplitHostPort(n.config.NtpRemoteServer)
+	if err != nil {
+		return err
+	}
+	portNum, err := strconv.Atoi(port)
+	if err != nil {
+		return err
+	}
+
+	local := time.Now().UTC().Unix()
+	ntp, err := ntpclient.GetNetworkTime(url, portNum)
+	if err != nil {
+		return err
+	}
+	remote := ntp.Unix()
+	if local != remote {
+		return fmt.Errorf("System time is out of sync, local:%x remote:%x", local, remote)
+	}
```

Comment:
Is there a reason two Unix timestamps are compared?
Unix timestamp may differ due to slight network delay. (e.g. local = 10.99s, remote = 11.01s )
How about comparing with some tolerance?
```go
var ntpTolerance = time.Second // choose whatever you need

func timeIsNear(lhs, rhs time.Time) bool {
        diff := lhs.Sub(rhs)
        if diff < 0 { // TODO: use time.Duration.Abs() after go1.19
                diff = -diff
        }
        return diff < ntpTolerance
}
```

Reply:
Great suggestion! resolved it. Thanks.